File: simulations-html/simulations/beers-law-lab/js/concentration/model/Beaker.js
// Copyright 2013, University of Colorado

define( [], function() {

    /**
     * Constructor
     * @param {Point2D} location bottom center
     * @param {Dimension2D} size
     * @param {Number} volume in liters (L)
     * @constructor
     */
    function Beaker( location, size, volume ) {
        this.location = location;
        this.size = size;
        this.volume = volume;
    }

    // Gets the x coordinate of the left wall.
    Beaker.prototype.getMinX = function () {
        return this.location.x - ( this.size.width / 2 );
    }

    return Beaker;
} );

File: branches/balancing-act-study-branch/simulations-java/simulations/beers-law-lab/src/edu/colorado/phet/beerslawlab/concentration/model/Beaker.java
// Copyright 2002-2012, University of Colorado
package edu.colorado.phet.beerslawlab.concentration.model;

import edu.colorado.phet.common.phetcommon.math.vector.Vector2D;
import edu.umd.cs.piccolo.util.PDimension;

/**
 * Model of an immutable beaker.
 *
 * @author Chris Malley (cmalley@pixelzoom.com)
 */
public class Beaker {

    public final Vector2D location; // bottom center
    public final PDimension size;
    public final double volume; // L

    public Beaker( Vector2D location, PDimension size, double volume ) {
        this.location = location;
        this.size = new PDimension( size );
        this.volume = volume;
    }

    // Gets the x coordinate of the left wall.
    public double getMinX() {
        return location.getX() - ( size.getWidth() / 2 );
    }
}
